.devcontainer/Dockerfile

```diff
@@ -0,0 +1,45 @@
+# [Choice] Go version: 1, 1.16, 1.15
+ARG VARIANT=1
+FROM mcr.microsoft.com/vscode/devcontainers/go:0-${VARIANT}
+
+# [Option] Install Node.js
+ARG DEBIAN_FRONTEND=noninteractive
+ARG INSTALL_NODE="true"
+ARG NODE_VERSION="lts/*"
+RUN if [ "${INSTALL_NODE}" = "true" ]; then su vscode -c "umask 0002 && . /usr/local/share/nvm/nvm.sh && nvm install ${NODE_VERSION} 2>&1"; fi \
+    # Install the Azure CLI
+    && echo "deb [arch=amd64] https://packages.microsoft.com/repos/azure-cli/ $(lsb_release -cs) main" > /etc/apt/sources.list.d/azure-cli.list \
+    && curl -sL https://packages.microsoft.com/keys/microsoft.asc | apt-key add - 2>/dev/null \
+    && apt-get update \
+    && apt-get install -y azure-cli \
+    # Install Docker CLI
+    && apt-get install -y gnupg-agent software-properties-common \
+    && curl -fsSL https://download.docker.com/linux/$(lsb_release -is | tr '[:upper:]' '[:lower:]')/gpg | (OUT=$(apt-key add - 2>&1) || echo $OUT) \
+    && add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/$(lsb_release -is | tr '[:upper:]' '[:lower:]') $(lsb_release -cs) stable" \
+    && apt-get update \
+    && apt-get install -y docker-ce-cli \
+    # Install Python
+    && apt-get update \
+    && apt-get install -y python3.6 python3-pip \
+    # Clean up
+    && apt-get autoremove -y && apt-get clean -y && rm -rf /var/lib/apt/lists/* /tmp/library-scripts
```

Comment:
Can you explain why these additional packages are required? These aren't included in the default Dockerfile that's created when adding a remote container https://github.com/microsoft/vscode-dev-containers/blob/v0.187.0/containers/go/.devcontainer/Dockerfile

Reply:
I removed the azure CLI package cause that's not necessary! Python is needed to install pre-commit, and Node is needed to install markdownlint. Shellcheck is added for linting bash scripts. And the docker CLI is now installed using the vscode repo's recommended install script as shown [here](https://github.com/microsoft/vscode-dev-containers/blob/main/script-library/docs/docker.md). This allows you to run docker commands from inside the container using a forwarded Docker socket. If you think that the dev container doesn't need Docker I can remove it!